### ISSUE TITLE
[Feature Request] Magical height: 'auto' for <Transition />

### ISSUE THREAD

**Issue by rockmandash** [drcmda/react-spring#153]
Hello, thank you for this wonderful library!
This library is amazing, I was trying height: 'auto' for <Spring />.
But I notice that <Transition /> seems not supporting this feature, it would be wonderful for supporting it, or other method for achieving <Transition /> + height: 'auto'.
Thank you very much!!!

**Comment by drcmda** [comment]
@rockmandash auto is supported

demo: https://codesandbox.io/embed/pmjomxn60

api: https://github.com/drcmda/react-spring/blob/master/API-OVERVIEW.md#animating-auto

where did you have problems?

**Comment by rockmandash** [comment]
@drcmda Thank you for your reply!
Sorry, I didn't make my question clear, my bad!
I was trying to do this:
```javascript
<Transition from={{ height: 0 }} enter={{ height: 'auto' }} leave={{ height: 0 }}>
  {toggle && Component}
</Transition>
``` 
I found out this didn't work, so I was wondering if <Transition /> can also support Magical height: 'auto', or is there any method to do this?
Thank you for your amazing library again!

**Comment by drcmda** [comment]
Transition internally is also just a wrap around spring, auto should work. For instance, try this example:

https://codesandbox.io/embed/j150ykxrv

and change `100` to `'auto'`

**Comment by rockmandash** [comment]
@drcmda Thank you!
Now I know I need more context, thank you!

**Comment by dvnrsn** [comment]
> When it doesn't work, it's probably because it needs more context, like parents positioning isn't absolute or relative, or the parent has zero height/width + overflow, something like that.

When I try height: "auto" on the enter, I get 
```
TypeError: Cannot read property 'props' of null
    at Object.fixAuto [as inject]```

Changing positioning on parent seems to have no effect and the parent does not have zero height. Any idea what I am misunderstanding?

**Comment by drcmda** [comment]
@dvnrsn i didn't check against the case where a spring returns undefined, it should be fixed now.

**Comment by dvnrsn** [comment]
Thanks, were module references changed in 5.5.2?
https://codesandbox.io/s/m86qjl8kp

**Comment by drcmda** [comment]
Yes, should be, but doesnt look too good there, i have no idea what "globals" is supposed to be. I'll take a look ...

**Comment by drcmda** [comment]
Dammit, something is seriously wrong - my macbook was dropping keys and i had it sent in, im on windows now and it seems to resolve files differently. :-S

**Comment by drcmda** [comment]
@dvnrsn now it should be fine: https://codesandbox.io/s/lpqkzk9137

I made some changes to your sandbox. you wouldn't see any height transition because there's no css overflow, with height set to 100 on the inner element, the outer element would simply flow over without any visual effect. I've also set it to absolute, because when you toggle it out it would take space, shifting a newly appearing element out of the way.

**Comment by drcmda** [comment]
@dvnrsn one more thing, your sandbox has react/react-dom 16.3.1 in there, everything before 16.4.1 had a sever bug in there regarding getDerivedStateFromProps, in react-springs case this would make exit-transition elements stick around. They've fixed it since then.

**Comment by wilcoschoneveld** [comment]
Thanks @drcmda for the amount of work you've already put into this library, I'm a huge fan.

I'm working on a tree component where I want to animate child nodes into view, and this can happen some levels deep. It seems that the outer 'auto' height transition does not respond well to the inner 'auto' height transition. Example: https://codesandbox.io/s/o75k1knl6q

Is this a bug of the framework or should I approach this differently?

**Comment by drcmda** [comment]
edit, looking at your example, i think this is something else. it doesn't calculate height properly, which most likely means it doesn't have enough context - let me check ...

**Comment by drcmda** [comment]
Ok, auto is definitively broken - again :-S react-animated-tree is down aswell with latest react-spring. Will fix.

**Comment by dvnrsn** [comment]
Agreed, appreciate all the work. @wilcoschoneveld thanks for the post. I was having this delayed nested height animation jump issue locally but was not able to replicate on codesandbox so I figured it was something with my code and gave up.

**Comment by drcmda** [comment]
@wilcoschoneveld @dvnrsn hang in there i'm almost done. i implemented some part of spring from scratch using react 16 lifecycles and trying to clean it up as best as i can in order to make the most complelx stuff like auto not such a headbreaker. I'll be fixed very soon.

**Comment by drcmda** [comment]
react-spring@5.6.0-beta2 (and newer, if available) should take are of "auto"

**Comment by wilcoschoneveld** [comment]
Cool! I couldn't get 5.6.1 to work with the example though; https://codesandbox.io/s/o75k1knl6q. Are there still issues?

**Comment by drcmda** [comment]
@wilcoschoneveld looks like it :( , i only checked the demos on the frontpage which started working again. auto is really messing me up these days ...

**Comment by DominicTobias** [comment]
Hi do you know why it doesn't work with v8? Thanks

Demo: https://codesandbox.io/s/3wwxnmw51

**Comment by trlmkb** [comment]
@DominicTobias not sure why, but if you remove attribute `native` from <Spring>, it seems to be working
